### ISSUE TITLE
Rule proposal: `no-identical-catch`

### ISSUE THREAD

**Issue by dimaMachina** [sindresorhus/eslint-plugin-unicorn#2046]
### Description

Same `catch` statement inside another `catch` statement can be safely removed

### Fail

```js
try {
  if (foo) {
    try {
      for await (const result of value) {
        handleResponse(result);
      }
      setIsFetching(false);
      setSubscription(null);
    } catch (error) {
      setIsFetching(false); // ❌ duplicates can be safely removed
      setResponse(formatError(error)); // ❌ duplicates can be safely removed
      setSubscription(null); // ❌ duplicates can be safely removed
    }
  } else {
    handleResponse(value);
  }
} catch (error) {
  setIsFetching(false);
  setResponse(formatError(error));
  setSubscription(null);
}
```

### Pass

```js
try {
  if (foo) {
    try {
      for await (const result of value) {
        handleResponse(result);
      }
      setIsFetching(false);
      setSubscription(null);
    } catch (error) {
      setIsFetching(false);
      setResponse(someAnotherFunction(error));
      setSubscription(null);
    }
  } else {
    handleResponse(value);
  }
} catch (error) {
  setIsFetching(false);
  setResponse(formatError(error));
  setSubscription(null);
}
```

### Additional Info

real-world example https://github.com/graphql/graphiql/pull/3033

**Comment by fisker** [comment]
What if `setResponse` is like

```js
function setResponse(error) {
  if (error instance of FooError) {
     throw new BarError()
  }

  if (error instance of BarError) {
     // ...
  }
}
```

**Comment by B2o5T** [comment]
@fisker possible but very rare case, a perfect case for eslint-disable comment

**Comment by fisker** [comment]
I'd say this example is also rare case ... 

Anyway, if this get accepted, as a reminder, should be carefully when there are flow control statements, including `return`/`break`/`continue`/`throw`.

**Comment by fisker** [comment]
Another possible rule proposal in this code.

```diff
    try {
+    try {
      // ...
+    } finally {
      setIsFetching(false);
+    }      
-      setSubscription(null);
    } catch (error) {
-      setIsFetching(false);
      setResponse(someAnotherFunction(error));
-      setSubscription(null);
    }
+   finally {
+      setSubscription(null);
+  }
```

(typing on cellphone, maybe bad indentation)

**Comment by sindresorhus** [comment]
I dont think we should add this. It is not generally safe to remove the inner catch, as duplicated handler statements can throw and the change can alter control flow. The motivating example is a good manual refactor, but a lint rule would either be unsafe or need too many special cases
